src/Module/Admin/Queue.php

```diff
@@ -30,17 +30,17 @@ public static function content()
 
 		// get jobs from the workerqueue table
 		if ($deferred) {
-			$condition = ["NOT `done` AND `next_try` > ?", DateTimeFormat::utcNow()];
+			$condition = ["NOT `done` AND `retrial` > ?", 0];
 			$sub_title = L10n::t('Inspect Deferred Worker Queue');
 			$info = L10n::t("This page lists the deferred worker jobs. This are jobs that couldn't be executed at the first time.");
 		} else {
-			$condition = ["NOT `done` AND `next_try` < ?", DateTimeFormat::utcNow()];
+			$condition = ["NOT `done` AND `retrial` = ?", 0];
 			$sub_title = L10n::t('Inspect Worker Queue');
 			$info = L10n::t('This page lists the currently queued worker jobs. These jobs are handled by the worker cronjob you\'ve set up during install.');
 		}
 
 		// @TODO Move to Model\WorkerQueue::getEntries()
-		$entries = DBA::select('workerqueue', ['id', 'parameter', 'created', 'priority'], $condition, ['order' => ['priority']]);
+		$entries = DBA::select('workerqueue', ['id', 'parameter', 'created', 'priority'], $condition, ['limit' => 999, 'order' => ['created']]);
```

Comment:
Plz add the `max_queue_show_limit` as a setting :)

Reply:
I don't think that we have to make everything configurable.

Reply:
Ok, but at least I'd add it as a constant for the class. So its clearly visible and has a short explanation :)

Reply:
Having the value separated from the place where it is in use does add some obscurity but no enhanced visibility, I think.

Reply:
The value of the variable is meaningless for the code itself. It says that we want to print a maximum amount of lines. But in case we want to change this maximum amount, it's easier to find such parameters at the start of the class as "constant" with the doc "this value changes maximum amount of printed lines" as inside a method, where the definition may not be standard ("limit", "maxAmount", 999). Or at least we have to search if the meaning of these limit "really" means limiting the whole amount which are printed (obviously because of "999", but not necessarily..).

Reply:
Since we use this value at two directly related lines I don't see a reason to put it somewhere else so that people have to search for that constant to finally see its value. I think that this is some abstraction that is only done because "we always do abstractions". When I wanted to see the value for the limit then I search for the query (mostly I do search for the used template first). Then I see the constant, then I have to search again for the constant. Even if I thought that this constant might be the wanted value, I had to look into the core to check if this really is the case.

I do use constants whenever there are values that are used allover the code, like the values for the network fields. For the same reason I unified the commands for the delivery process and put them into appropriate constants. I prefer constants when they are used like Enums.

Reply:
You know exactly where to search because you're the author of the query. It's not some kind of abstraction, its based on best practices to help other developers finding faster necessary things without the need to reverse engineering the code.

Constants are not just repetitive used values, they give a static values a meaning and therefore is a good way to document code too. For example why do you choose 999? I think because 999 is assumed that gives the best performance during page refresh without losing too much information of the queue. These are implicit knowledges which are good to have documented too. 'What was the intention of the author?'

Reply:
🍿

Reply:
But I'm sorry, I don't want to bother you.. Let's just finish this PR before @MrPetovan is finished eating his popcorn :P

Reply:
I agree with @nupplaphil here, it's more about documentation than pure convenience. Think about yourself in 6-12 months, you won't necessarily remember why you chose something as specific as 999 for this query.

Reply:
I didn't knew why I picked 999 even when I wrote that code.

Reply:
It's even worse!

Reply:
Yeah, that really drove one of my bosses crazy when he asked why I coded something this or that way and I told: "Don't know, I felt that way." I mostly code without much thinking about it, driven mostly by intuition.